### PR TITLE
fix: Use -f with yaml of helm patch

### PR DESCRIPTION
**Pull request by mmorhun**
Signed-off-by: Mykola Morhun <mmorhun@redhat.com>

### What does this PR do?
Passes file provided by `--helm-patch-yaml` flag as a `values.yaml` parameter for Helm Chart.
Provided command line arguments oc chectl take precedence over the patch values.

### Screenshot/screencast of this PR
N/A

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/18459

### How to test this PR?
1. Create `helm-patch.yaml` file with some corrections, for example:
```yaml
global:
  cheWorkspacesNamespace: che-ws-<username>
```
2. Deploy Eclipse Che using Helm installer with the patch:
```sh
chectl server:deploy --platform=minikube --installer=helm --multiuser --helm-patch-yaml=helm-patch.yaml
```
3. Check that patch is applied (start a workspace and check the k8s namespace of it)

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by mmorhun** [comment]
@cccs-eric do you have any concerns/suggestions?

File: src/tasks/installers/helm.ts [hunk 1]
```diff
@@ -348,18 +348,9 @@ error: E_COMMAND_FAILED`)
...
-      const patchProperties = this.prepareYamlPatch(patchTree)
-      // Add patch properties to the command
-      for (const property of patchProperties) {
-        setOptions.push(`--set ${property}`)
-      }
-    }
+    const patchFlags = flags['helm-patch-yaml'] ? '-f ' + flags['helm-patch-yaml'] : ''
 
-    let command = `helm upgrade --install che --force --namespace ${flags.chenamespace} ${setOptions.join(' ')} ${multiUserFlag} ${tlsFlag} ${destDir}`
+    let command = `helm upgrade --install che --force --namespace ${flags.chenamespace} ${setOptions.join(' ')} ${multiUserFlag} ${tlsFlag} ${patchFlags} ${destDir}`
```
Comment (tolusha) [inline src/tasks/installers/helm.ts:353]: should `patchFlags` be at the very end ?

Comment (cccs-eric) [reply src/tasks/installers/helm.ts:353]:
```
Usage:
  helm upgrade [RELEASE] [CHART] [flags]
```
`flags` can be set at the end, or before (like `chectl` is doing).  Both will work.  In this case `${destDir}` is the `CHART` parameter.  If you want to follow helm's documented usage, you will need to shuffle more than that parameter.

**Comment by cccs-eric** [comment]
@mmorhun 
Code looks good, but I'll raison one point about your PR description.  According to [Helm's documentation](https://helm.sh/docs/chart_template_guide/values_files/), the `--set <arg>` will have precedence over the values file.

> ... values.yaml is the default, which can be overridden by a parent chart's values.yaml, which can in turn be overridden by a user-supplied values file, which can in turn be overridden by --set parameters.

So when you say

> Now, it passes provided yaml patch to helm directly via -f option in the end of the command, so the patch has the highest priority (as it is already done for operator patch).

it is not true since the `--set ` will overwrite the values file.  If this is your intent, it's all good.  Otherwise, you will have to create a YAML file with all the values set by `--set <key>=<value>` and put it before the helm patch file.  Should `chectl` configurations take over the patch file?  I'll leave that call to you guys! 😄

**Comment by tolusha** [comment]
At least we should be clear in descriptions.

File: README.md [hunk 2]
```diff
@@ -524,7 +524,9 @@ OPTIONS
       The URL of the external Devfile registry.
 
   --helm-patch-yaml=helm-patch-yaml
-      Path to yaml file with Helm Chart values patch. The file format is identical to values.yaml from the chart.
+      Path to yaml file with Helm Chart values patch.
+                           The file format is identical to values.yaml from the chart.
+                           Provided command line arguments take precedence.
```
Comment (tolusha) [inline README.md:529]: `NOTE: .. take precedence over patch file.`

**Review by tolusha — APPROVED** [review]

**Review by AndrienkoAleksandr — APPROVED** [review]

**Pull request merged**